Startup Crash with SpongeForge 1.16.5 in All The Mods 6 Server

**Describe the bug**
ElementalCraft no longer works with SpongeForge 1.16.5

```log
[24Mar2022 06:33:02.245] [main/INFO] [cpw.mods.modlauncher.Launcher/MODLAUNCHER]: ModLauncher running: args [--gameDir, ., --launchTarget, fmlserver, --fml.forgeVersion, 36.2.33, --fml.mcpVersion, 20210115.111550, --fml.mcVersion, 1.16.5, --fml.forgeGroup, net.minecraftforge, nogui]
[24Mar2022 06:33:02.248] [main/INFO] [cpw.mods.modlauncher.Launcher/MODLAUNCHER]: ModLauncher 8.1.3+8.1.3+main-8.1.x.c94d18ec starting: java version 1.8.0_312 by Private Build
[24Mar2022 06:33:02.430] [main/INFO] [net.minecraftforge.fml.loading.FixSSL/CORE]: Added Lets Encrypt root certificates as additional trust
[24Mar2022 06:33:02.461] [main/INFO] [mixin/]: SpongePowered MIXIN Subsystem Version=0.8.4 Source=file:/home/minecraft/atm6/libraries/org/spongepowered/mixin/0.8.4/mixin-0.8.4.jar Service=ModLauncher Env=SERVER
[24Mar2022 06:33:03.232] [main/INFO] [org.spongepowered.forge.applaunch.loading.moddiscovery.library.LibraryManager/]: Scanning and verifying libraries in './sponge-libraries'. Please wait, this may take a moment...
[24Mar2022 06:33:04.636] [main/INFO] [STDERR/]: [org.antlr.v4.runtime.ConsoleErrorListener:syntaxError:38]: line 13:0 token recognition error at: '`'
[24Mar2022 06:33:04.747] [main/INFO] [STDERR/]: [org.antlr.v4.runtime.ConsoleErrorListener:syntaxError:38]: line 1:0 token recognition error at: '~'
[24Mar2022 06:33:05.984] [main/INFO] [mixin/]: Successfully loaded Mixin Connector [shetiphian.core.mixins.MixinConnector]
[24Mar2022 06:33:05.985] [main/INFO] [mixin/]: Successfully loaded Mixin Connector [vazkii.botania.common.MixinConnector]
[24Mar2022 06:33:05.987] [main/INFO] [mixin/]: Successfully loaded Mixin Connector [com.thevortex.allthetweaks.mixin.MixinConnector]
[24Mar2022 06:33:05.988] [main/INFO] [mixin/]: Successfully loaded Mixin Connector [com.leobeliik.extremesoundmuffler.MixinConnector]
[24Mar2022 06:33:05.990] [main/INFO] [mixin/]: Successfully loaded Mixin Connector [vazkii.patchouli.common.MixinConnector]
[24Mar2022 06:33:05.993] [main/INFO] [mixin/]: Successfully loaded Mixin Connector [eutros.framedcompactdrawers.MixinConnector]
[24Mar2022 06:33:06.003] [main/INFO] [cpw.mods.modlauncher.LaunchServiceHandler/MODLAUNCHER]: Launching target 'fmlserver' with arguments [--gameDir, ., nogui]
[24Mar2022 06:33:06.088] [main/WARN] [mixin/]: Reference map 'findme.refmap.json' for findme.mixins.json could not be read. If this is a development environment you can ignore this message
[24Mar2022 06:33:06.099] [main/WARN] [mixin/]: Reference map 'modid.refmap.json' for rsinfinitybooster.mixins.json could not be read. If this is a development environment you can ignore this message
[24Mar2022 06:33:06.123] [main/WARN] [mixin/]: Reference map 'rhino-forge-refmap.json' for rhino.mixins.json could not be read. If this is a development environment you can ignore this message
[24Mar2022 06:33:06.133] [main/WARN] [mixin/]: Reference map 'ponderjs.refmap.json' for ponderjs.mixins.json could not be read. If this is a development environment you can ignore this message
[24Mar2022 06:33:07.189] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'add_custom_entity_collision' ASM patch...
[24Mar2022 06:33:07.214] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'add_custom_entity_collision' ASM patch!
[24Mar2022 06:33:07.350] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'reach_set_server_entity_interact' ASM patch...
[24Mar2022 06:33:07.359] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'reach_set_server_entity_interact' ASM patch!
[24Mar2022 06:33:07.364] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'allow_entity_interaction' ASM patch...
[24Mar2022 06:33:07.377] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'allow_entity_interaction' ASM patch!
[24Mar2022 06:33:07.495] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching CrossbowItem#onItemRightClick
[24Mar2022 06:33:07.506] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching CrossbowItem#fireProjectile
[24Mar2022 06:33:07.533] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching Item#getItemEnchantability
[24Mar2022 06:33:07.581] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'set_player_field' ASM patch...
[24Mar2022 06:33:07.583] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'set_player_field' ASM patch!
[24Mar2022 06:33:07.589] [main/INFO] [net.minecraftforge.coremod.CoreMod.placebo/COREMODLOG]: Patching LivingEntity#attackEntityFrom
[24Mar2022 06:33:07.610] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching LivingEntity#blockUsingShield
[24Mar2022 06:33:07.614] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching LivingEntity#applyPotionDamageCalculations
[24Mar2022 06:33:07.619] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'water_movement_slowdown_prevention' ASM patch...
[24Mar2022 06:33:07.621] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'water_movement_slowdown_prevention' ASM patch!
[24Mar2022 06:33:07.735] [main/INFO] [net.minecraftforge.coremod.CoreMod.placebo/COREMODLOG]: Patching ItemStack#onItemUse
[24Mar2022 06:33:07.740] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'add_missing_tag_enchantment_tooltip' ASM patch...
[24Mar2022 06:33:07.752] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'add_missing_tag_enchantment_tooltip' ASM patch!
[24Mar2022 06:33:07.755] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'add_enchantment_tooltip' ASM patch...
[24Mar2022 06:33:07.761] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'add_enchantment_tooltip' ASM patch!
[24Mar2022 06:33:07.820] [main/WARN] [mixin/]: Error loading class: de/maxhenkel/gravestone/events/DeathEvents (java.lang.ClassNotFoundException: null)
[24Mar2022 06:33:07.823] [main/WARN] [mixin/]: Error loading class: de/maxhenkel/corpse/events/DeathEvents (java.lang.ClassNotFoundException: null)
[24Mar2022 06:33:07.839] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching net/minecraft/loot/functions/EnchantRandomly
[24Mar2022 06:33:07.937] [main/INFO] [net.minecraftforge.coremod.CoreMod.observerlib/COREMODLOG]: Adding 'on_block_change' ASM patch...
[24Mar2022 06:33:07.940] [main/INFO] [net.minecraftforge.coremod.CoreMod.observerlib/COREMODLOG]: Added 'on_block_change' ASM patch!
[24Mar2022 06:33:07.942] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'sun_brightness_server' ASM patch...
[24Mar2022 06:33:07.944] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'sun_brightness_server' ASM patch!
[24Mar2022 06:33:08.124] [main/FATAL] [net.minecraftforge.fml.loading.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/gui/screen/inventory/ContainerScreen for invalid dist DEDICATED_SERVER
[24Mar2022 06:33:08.124] [main/WARN] [mixin/]: Error loading class: net/minecraft/client/gui/screen/inventory/ContainerScreen (java.lang.RuntimeException: Attempted to load class net/minecraft/client/gui/screen/inventory/ContainerScreen for invalid dist DEDICATED_SERVER)
[24Mar2022 06:33:08.124] [main/WARN] [mixin/]: @Mixin target net.minecraft.client.gui.screen.inventory.ContainerScreen was not found findme.mixins.json:MixinSlotRenderer
[24Mar2022 06:33:08.195] [main/FATAL] [net.minecraftforge.fml.loading.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/entity/player/ClientPlayerEntity for invalid dist DEDICATED_SERVER
[24Mar2022 06:33:08.195] [main/WARN] [mixin/]: Error loading class: net/minecraft/client/entity/player/ClientPlayerEntity (java.lang.RuntimeException: Attempted to load class net/minecraft/client/entity/player/ClientPlayerEntity for invalid dist DEDICATED_SERVER)
[24Mar2022 06:33:08.195] [main/WARN] [mixin/]: @Mixin target net.minecraft.client.entity.player.ClientPlayerEntity was not found ars_nouveau.mixins.json:ClientElytraMixin
[24Mar2022 06:33:08.344] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching Item#getItemEnchantability
[24Mar2022 06:33:08.419] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching RepairContainer#updateRepairOutput
[24Mar2022 06:33:08.433] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Successfully removed the anvil level cap.
[24Mar2022 06:33:08.439] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching RepairContainer#updateRepairOutput
[24Mar2022 06:33:08.451] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Successfully allowed nbt-unbreakable items in the anvil.
[24Mar2022 06:33:08.452] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching net/minecraft/inventory/container/RepairContainer
[24Mar2022 06:33:08.469] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'add_enchantment_helper_levels' ASM patch...
[24Mar2022 06:33:08.473] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'add_enchantment_helper_levels' ASM patch!
[24Mar2022 06:33:08.476] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'add_enchantment_helper_enchantments' ASM patch...
[24Mar2022 06:33:08.478] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'add_enchantment_helper_enchantments' ASM patch!
[24Mar2022 06:33:08.481] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'add_enchantment_helper_apply_modifier' ASM patch...
[24Mar2022 06:33:08.483] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'add_enchantment_helper_apply_modifier' ASM patch!
[24Mar2022 06:33:08.487] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching EnchantmentHelper#getEnchantmentModifierDamage
[24Mar2022 06:33:08.493] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching EnchantmentHelper#getModifierForCreature
[24Mar2022 06:33:08.499] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching EnchantmentHelper#applyThornEnchantments
[24Mar2022 06:33:08.529] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching EnchantmentHelper#applyArthropodEnchantments
[24Mar2022 06:33:08.536] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching buildEnchantmentList for the Enchantability affix.
[24Mar2022 06:33:08.542] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching EnchantmentHelper#getEnchantmentDatas
[24Mar2022 06:33:08.611] [main/INFO] [net.minecraftforge.coremod.CoreMod.observerlib/COREMODLOG]: Adding 'on_block_change' ASM patch...
[24Mar2022 06:33:08.611] [main/INFO] [net.minecraftforge.coremod.CoreMod.observerlib/COREMODLOG]: Added 'on_block_change' ASM patch!
[24Mar2022 06:33:08.612] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'sun_brightness_server' ASM patch...
[24Mar2022 06:33:08.612] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'sun_brightness_server' ASM patch!
[24Mar2022 06:33:09.028] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching CampfireTileEntity#findMatchingRecipe
[24Mar2022 06:33:09.846] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'post_process_vanilla' ASM patch...
[24Mar2022 06:33:09.848] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'post_process_vanilla' ASM patch!
[24Mar2022 06:33:09.856] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching SharedMonsterAttributes#readAttributeModifier
[24Mar2022 06:33:10.164] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching FishingBobberEntity#catchingFish
[24Mar2022 06:33:10.239] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'cooldown_tracker_set' ASM patch...
[24Mar2022 06:33:10.241] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'cooldown_tracker_set' ASM patch!
[24Mar2022 06:33:11.671] [main/INFO] [net.minecraftforge.coremod.CoreMod.placebo/COREMODLOG]: Patching DataPackRegistries#<init>
[24Mar2022 06:33:11.676] [main/INFO] [net.minecraftforge.coremod.CoreMod.placebo/COREMODLOG]: Patching LootTableManager#apply
[24Mar2022 06:33:11.789] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching TemptGoal#isTempting
[24Mar2022 06:33:13.304] [main/INFO] [com.unrealdinnerbone.delogger.LoggerHacks/]: !!! SOME LOGGERS HAVE MOVED TO THE DEBUG FILE (Please Include them in your bug reports) !!!
[24Mar2022 06:33:16.765] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'set_player_field' ASM patch...
[24Mar2022 06:33:16.766] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'set_player_field' ASM patch!
[24Mar2022 06:33:16.766] [main/INFO] [net.minecraftforge.coremod.CoreMod.placebo/COREMODLOG]: Patching LivingEntity#attackEntityFrom
[24Mar2022 06:33:16.770] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching LivingEntity#blockUsingShield
[24Mar2022 06:33:16.770] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching LivingEntity#applyPotionDamageCalculations
[24Mar2022 06:33:16.771] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'water_movement_slowdown_prevention' ASM patch...
[24Mar2022 06:33:16.771] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'water_movement_slowdown_prevention' ASM patch!
[24Mar2022 06:33:17.025] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'add_custom_entity_collision' ASM patch...
[24Mar2022 06:33:17.025] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'add_custom_entity_collision' ASM patch!
[24Mar2022 06:33:17.853] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching RepairContainer#updateRepairOutput
[24Mar2022 06:33:17.855] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Successfully removed the anvil level cap.
[24Mar2022 06:33:17.856] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching RepairContainer#updateRepairOutput
[24Mar2022 06:33:17.858] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Successfully allowed nbt-unbreakable items in the anvil.
[24Mar2022 06:33:17.859] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching net/minecraft/inventory/container/RepairContainer
[24Mar2022 06:33:18.264] [main/INFO] [net.minecraftforge.coremod.CoreMod.apotheosis/COREMODLOG]: Patching Item#getItemEnchantability
[24Mar2022 06:33:19.282] [main/INFO] [net.minecraftforge.coremod.CoreMod.observerlib/COREMODLOG]: Adding 'on_block_change' ASM patch...
[24Mar2022 06:33:19.282] [main/INFO] [net.minecraftforge.coremod.CoreMod.observerlib/COREMODLOG]: Added 'on_block_change' ASM patch!
[24Mar2022 06:33:19.282] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Adding 'sun_brightness_server' ASM patch...
[24Mar2022 06:33:19.282] [main/INFO] [net.minecraftforge.coremod.CoreMod.astralsorcery/COREMODLOG]: Added 'sun_brightness_server' ASM patch!
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:39)
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54)
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72)
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82)
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66)
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.server.ServerMain$Runner.runLauncher(ServerMain.java:63)
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.server.ServerMain$Runner.access$100(ServerMain.java:60)
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.server.ServerMain.main(ServerMain.java:57)
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1061]: Caused by: java.lang.reflect.InvocationTargetException
[24Mar2022 06:33:19.864] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1061]: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1061]: 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1061]: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1061]: 	at java.lang.reflect.Method.invoke(Method.java:498)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1061]: 	at net.minecraftforge.fml.loading.FMLServerLaunchProvider.lambda$launchService$0(FMLServerLaunchProvider.java:51)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1061]: 	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:1061]: 	... 7 more
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:250)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at org.spongepowered.asm.service.modlauncher.MixinTransformationHandler.processClass(MixinTransformationHandler.java:131)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at org.spongepowered.asm.launch.MixinLaunchPluginLegacy.processClass(MixinLaunchPluginLegacy.java:131)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at cpw.mods.modlauncher.serviceapi.ILaunchPluginService.processClassWithFlags(ILaunchPluginService.java:154)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:85)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120)
[24Mar2022 06:33:19.865] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:265)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:136)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:98)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.block.Blocks.<clinit>(Blocks.java:30)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.world.gen.surfacebuilders.SurfaceBuilder.<clinit>(SurfaceBuilder.java:12)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.world.gen.surfacebuilders.ConfiguredSurfaceBuilders.<clinit>(SourceFile:11)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.util.registry.WorldGenRegistries.func_243674_g(WorldGenRegistries.java:33)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.util.registry.WorldGenRegistries.func_243668_a(WorldGenRegistries.java:89)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.util.registry.WorldGenRegistries.<clinit>(WorldGenRegistries.java:88)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.util.registry.Registry.<clinit>(Registry.java:464)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.util.registry.Bootstrap.func_151354_b(Bootstrap.java:38)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	at net.minecraft.server.Main.main(Main.java:92)
[24Mar2022 06:33:19.866] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:644]: 	... 13 more
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Variable modifier method attackTargetEntityWithCurrentItemSetFlag3(Z)Z in elementalcraft.mixins.json:MixinPlayerEntity failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap elementalcraft.refmap.json
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468)
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1362)
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1051)
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400)
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325)
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:383)
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:365)
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
[24Mar2022 06:33:19.867] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:668]: 	... 33 more

```

**ModPack**
ATM6 is the Modpack

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,9 +1,9 @@
 minecraft_version=1.16.5
-loader_version=36.0
+loader_version=36.2
 mappings_channel=official
 mappings_version=1.16.5
 forge_version=1.16.5-36.0.17
-mixins_version=0.8.2
+mixins_version=0.8.4
 
 curse_release_type=release
 
@@ -12,7 +12,7 @@ elementalcraft_version=1.16.5-2.7.14
 # mod dependencies
 jei_version=1.16.5:7.7.0.106
 patchouli_version=1.16.4-50
-dpanvil_version=1.16.5-1.2.2
+dpanvil_version=1.16.5-1.3.1
 
 # mod interactions
 mekanism_version=1.16.3-10.0.15.441

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,7 +3,7 @@ loader_version=36.2
 mappings_channel=official
 mappings_version=1.16.5
 forge_version=1.16.5-36.2.20
-mixins_version=0.8.4
+mixins_version=0.8.2
 
 curse_release_type=release
 

--- a/gradle.properties
+++ b/gradle.properties
@@ -2,17 +2,17 @@ minecraft_version=1.16.5
 loader_version=36.2
 mappings_channel=official
 mappings_version=1.16.5
-forge_version=1.16.5-36.0.17
+forge_version=1.16.5-36.2.20
 mixins_version=0.8.4
 
 curse_release_type=release
 
-elementalcraft_version=1.16.5-2.7.14
+elementalcraft_version=1.16.5-2.7.17
 
 # mod dependencies
-jei_version=1.16.5:7.7.0.106
-patchouli_version=1.16.4-50
+jei_version=1.16.5:7.7.1.152
+patchouli_version=1.16.4-53.2
 dpanvil_version=1.16.5-1.3.1
 
 # mod interactions
-mekanism_version=1.16.3-10.0.15.441
+mekanism_version=1.16.5-10.1.2.457

--- a/src/main/java/sirttas/elementalcraft/mixin/MixinPlayerEntity.java
+++ b/src/main/java/sirttas/elementalcraft/mixin/MixinPlayerEntity.java
@@ -21,15 +21,18 @@ public abstract class MixinPlayerEntity extends LivingEntity {
 		super(type, worldIn);
 	}
 
-	@ModifyVariable(method = "attack(Lnet/minecraft/entity/Entity;)V", 
-			index = 10, 
-			at = @At(value = "INVOKE_ASSIGN", target = "Lnet/minecraft/item/ItemStack;getItem()Lnet/minecraft/item/Item;", shift = Shift.AFTER))
-	public boolean attackTargetEntityWithCurrentItemSetFlag3(boolean flag3) {
-		if (isHoldinStaff()) {
-			return true;
-		}
-		return flag3;
-	}
+	@SuppressWarnings("rawtypes")
+        @Redirect(
+        method = "attack(Lnet/minecraft/world/entity/Entity;)V",
+            at = @At(
+                value = "INVOKE",
+                target = "Lnet/minecraft/item/ItemStack;getItem()Lnet/minecraft/item/Item;",
+                shift = At.Shift.AFTER
+            )
+        )
+        private boolean impl$onAttack(final Object item, final Class target) {
+            return item instanceof SwordItem || isHoldinStaff();
+        }
 	
 	@Redirect(method = "attack(Lnet/minecraft/entity/Entity;)V", 
 			at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/Entity;getBoundingBox()Lnet/minecraft/util/math/AxisAlignedBB;"))

--- a/src/main/java/sirttas/elementalcraft/mixin/MixinPlayerEntity.java
+++ b/src/main/java/sirttas/elementalcraft/mixin/MixinPlayerEntity.java
@@ -1,9 +1,8 @@
 package sirttas.elementalcraft.mixin;
 
+import net.minecraft.item.SwordItem;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
-import org.spongepowered.asm.mixin.injection.At.Shift;
-import org.spongepowered.asm.mixin.injection.ModifyVariable;
 import org.spongepowered.asm.mixin.injection.Redirect;
 
 import net.minecraft.entity.Entity;
@@ -22,17 +21,18 @@ public abstract class MixinPlayerEntity extends LivingEntity {
 	}
 
 	@SuppressWarnings("rawtypes")
-        @Redirect(
-        method = "attack(Lnet/minecraft/world/entity/Entity;)V",
-            at = @At(
-                value = "INVOKE",
-                target = "Lnet/minecraft/item/ItemStack;getItem()Lnet/minecraft/item/Item;",
-                shift = At.Shift.AFTER
-            )
-        )
-        private boolean impl$onAttack(final Object item, final Class target) {
-            return item instanceof SwordItem || isHoldinStaff();
-        }
+	@Redirect(
+			method = "attack(Lnet/minecraft/entity/Entity;)V",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/item/ItemStack;getItem()Lnet/minecraft/item/Item;",
+					shift = At.Shift.AFTER
+			)
+	)
+
+	private boolean impl$onAttack(final Object item, final Class target) {
+		return item instanceof SwordItem || isHoldinStaff();
+	}
 	
 	@Redirect(method = "attack(Lnet/minecraft/entity/Entity;)V", 
 			at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/Entity;getBoundingBox()Lnet/minecraft/util/math/AxisAlignedBB;"))
@@ -46,5 +46,4 @@ public abstract class MixinPlayerEntity extends LivingEntity {
 	private boolean isHoldinStaff() {
 		return this.getMainHandItem().getItem() instanceof StaffItem;
 	}
-	
 }

